File: python/syntax.py
#print “Hello World” on the console

print("Hello world")

#SYNTAX:
#define attributes / variables

#numbers
a = 1
b = 1.0

#string
stringVar1 = "this is a string"
stringVar2 = 'this is a string'

#boolean
booleanVar1 = True
booleanVar2 = False

#arrays 
pythonArray = ["XT", "MR2", "AW11", "GK5"]

#dictionary
pythonDict = {
	'maker': "Toyota",
	'model': "MR2",
	'code': "AW11",
	'engine': "4A-GE",
}

#sample if / else
a = 2
if (a == 10 or a > 10):
    print('the number is not smaller then 10')
else: 
	print('the number is smaller then 10')

# functions
# 	- parameters
# 	- returns 
def exampleFunction(b):
	if (b == 10 or b > 10):
		print("number is  NOT smaller then 10")
		return b
	else:
		print("WELL DONE!!!!")
		return b

exampleFunction(1)

# arrays
arr = [1,2,3,4,5,6]
# 	- add to the front
arr.insert(0,'a')
# 	- add to the end
arr.append(7)
arr.insert(len(arr),8)
# 	- update values
arr[0] = 0

# loops - for
for element in arr: 
    print(element)

# loops - while
i=0
while i < len(arr):
    print(f'arr[{i}]={arr[i]}')
    i=i+1

File: python/syntax.js

// define attributes / variables
//  - number
//  - string
//  - boolean
//  - array
//  - dictionary / objects 
//  - undefined 

var globalVar = 10;
let blockVar = 2;
console.log (globalVar);

var stringVar = "string";
console.log (stringVar);

var booleanVar = true;
console.log (booleanVar);

var arrayVar = ["XT", "MR2", "AW11", "GK5"];
console.log (arrayVar);

var objectVar = {
	maker: "Toyota",
	model: "MR2",
	code: "AW11",
	engine: "4A-GE",
}
console.log (objectVar);

var unidentifiedVar;
console.log (unidentifiedVar);

// sample if / else
var numberVar = prompt("enter a number smaller then 10");
if (numberVar === 10 || numberVar > 10){
	alert ("YOU ARE WRONG!!!! your number is  NOT smaller then 10");
} else {
	alert ("WELL DONE!!!!  (you can follow orders!)");
}

// functions
// 	- parameters
// 	- returns 
alert ("let's try again...");
var numberVar2 = prompt("enter a number smaller then 10");

function exampleFunction(a,b) {
	if (numberVar2 === globalVar || numberVar2 > globalVar){
		alert ("YOU ARE WRONG!!!! your number is  NOT smaller then 10");
		return numberVar2;
	} else {
		alert ("WELL DONE!!!!  (you can follow orders!)");
		return numberVar2;
	}	
}
exampleFunction (numberVar2,globalVar);

// arrays
// 	- add to the front
// 	- add to the end
// 	- update values
var newElement = prompt("type a word to add at the begining of the array: [" + arrayVar + "]");
arrayVar.unshift(newElement);
console.log (arrayVar);

var newElement = prompt ("type a word to add at the begining of the array: [" + arrayVar + "]");
arrayVar.push(newElement);
console.log (arrayVar);


// loops - for
function forLoopFunct(num,power) {
	let a = 1;
	for (i=0; i<power; i++){
		a = a * num;
	}
	return a;
}

console.log (forLoopFunct(2,6));

// loops - for/in
function forInLoopFunct (obj){
	let str = "";
	for (var properties in obj) {
		str = str + " " + obj[properties];
	}
	return str;
}
console.log (forInLoopFunct(objectVar));

// Loops- while
function whileLoopFunct(num,power) {
	let i = 0;
	let a = 1;
	while (i < power) {
		a = a * num;
		i++;
	}
	return a;
}
console.log (whileLoopFunct(2,6));

// Loops- do while
function doWhileLoopFunct(num,power) {
	let i = 0;
	let a = 1;
	do {
		a = a * num;
		i++;
	}
	while (i < power);
	return a;
}
console.log (doWhileLoopFunct(2,6));

// 	- forEach (with array and function) -->
arrayVar.forEach(function(x) {
	console.log(x);
});



//Objects / Dictionaries
	// - declare object
	// - lookup key to retrieve value
console.log (objectVar.model);